onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -5190,8 +5191,361 @@ TEST_F(GraphTransformationTests, PropagateCastOpsTests_Gelu) {
                          pre_graph_checker, post_graph_checker);
   }
 }
+
 #endif
 
+void CheckShapeEquality(const ONNX_NAMESPACE::TensorShapeProto* shape1,
+                        const ONNX_NAMESPACE::TensorShapeProto* shape2) {
+  EXPECT_NE(shape1, nullptr);
+  EXPECT_NE(shape2, nullptr);
+  if ((shape1 != nullptr) && (shape2 != nullptr)) {
+    EXPECT_EQ(shape1->dim_size(), shape2->dim_size()) << "Shapes do not have same rank";
+    auto min_dims = std::min(shape1->dim_size(), shape2->dim_size());
+    for (int i = 0; i < min_dims; ++i) {
+      auto dim1 = shape1->dim(i);
+      auto dim2 = shape2->dim(i);
+      EXPECT_EQ(dim1.has_dim_value(), dim2.has_dim_value());
+      if (dim1.has_dim_value()) {
+        EXPECT_EQ(dim1.dim_value(), dim2.dim_value());
+      }
+      EXPECT_EQ(dim1.has_dim_param(), dim2.has_dim_param());
+      if (dim1.has_dim_param()) {
+        EXPECT_EQ(dim1.dim_param(), dim2.dim_param());
+      }
+    }
+  }
+}
+
+TEST_F(GraphTransformationTests, ConstantSharing_AddClip) {
```

Comment:
nit: for each test please add a comment saying what's it's testing so the reader doesn't have to decipher the implementation to figure it out. Or make the test name a lot more descriptive.

Reply:
Added more descriptive comments, hope they make sense. Thank you for the suggestion!!